From c7ed09bab41d0a1c0c23db3eeafcd974b94221ab Mon Sep 17 00:00:00 2001
From: David Hampton <mythtv@love2code.net>
Date: Fri, 24 Dec 2021 16:28:27 -0500
Subject: [PATCH] Flip the ALLOWLASTPLAYPOS flag to become the
 IGNORELASTPLAYPOS flag.

This brings the flag into line with the other IGNORExxx flags. Given
that the last play position is never set, this change should have no
effect on the code.
---
 mythtv/libs/libmyth/programinfo.cpp                  |  2 +-
 mythtv/libs/libmyth/programinfo.h                    |  6 +++---
 mythtv/libs/libmyth/programtypeflags.h               |  2 +-
 .../libmyth/test/test_programinfo/test_programinfo.h |  4 ++--
 mythtv/libs/libmythtv/mythplayerui.cpp               |  2 +-
 mythtv/libs/libmythtv/previewgenerator.cpp           |  2 +-
 mythtv/libs/libmythtv/tv_play.cpp                    |  2 +-
 mythtv/libs/libmythtv/tv_play.h                      | 12 ++++++------
 mythtv/programs/mythfrontend/playbackbox.cpp         | 10 +++++-----
 9 files changed, 21 insertions(+), 21 deletions(-)

diff --git a/mythtv/libs/libmyth/programinfo.cpp b/mythtv/libs/libmyth/programinfo.cpp
index a0706fe66c0..4446da35901 100644
--- a/mythtv/libs/libmyth/programinfo.cpp
+++ b/mythtv/libs/libmyth/programinfo.cpp
@@ -2796,7 +2796,7 @@ uint64_t ProgramInfo::QueryProgStart(void) const
  */
 uint64_t ProgramInfo::QueryLastPlayPos(void) const
 {
-    if (!(m_programFlags & FL_ALLOWLASTPLAYPOS))
+    if (m_programFlags & FL_IGNORELASTPLAYPOS)
         return 0;
 
     frm_dir_map_t bookmarkmap;
diff --git a/mythtv/libs/libmyth/programinfo.h b/mythtv/libs/libmyth/programinfo.h
index 07578946969..dfc963972dd 100644
--- a/mythtv/libs/libmyth/programinfo.h
+++ b/mythtv/libs/libmyth/programinfo.h
@@ -571,10 +571,10 @@ class MPUBLIC ProgramInfo
     /// \brief If "ignore" is true QueryLastPlayPos() will return 0, otherwise
     ///        QueryLastPlayPos() will return the last playback position
     ///        if it exists.
-    void SetAllowLastPlayPos(bool allow)
+    void SetIgnoreLastPlayPos(bool ignore)
     {
-        m_programFlags &= ~FL_ALLOWLASTPLAYPOS;
-        m_programFlags |= (allow) ? FL_ALLOWLASTPLAYPOS : 0;
+        m_programFlags &= ~FL_IGNORELASTPLAYPOS;
+        m_programFlags |= (ignore) ? FL_IGNORELASTPLAYPOS : 0;
     }
     virtual void SetRecordingID(uint _recordedid)
         { m_recordedId = _recordedid; }
diff --git a/mythtv/libs/libmyth/programtypeflags.h b/mythtv/libs/libmyth/programtypeflags.h
index 0c10311e8dd..d7e2d93a177 100644
--- a/mythtv/libs/libmyth/programtypeflags.h
+++ b/mythtv/libs/libmyth/programtypeflags.h
@@ -53,7 +53,7 @@ FLAGS_DATA(FL, DUPLICATE,        0x00002000)
 FLAGS_DATA(FL, REACTIVATE,       0x00004000)
 FLAGS_DATA(FL, IGNOREBOOKMARK,   0x00008000)
 FLAGS_DATA(FL, IGNOREPROGSTART,  0x00010000)
-FLAGS_DATA(FL, ALLOWLASTPLAYPOS, 0x00020000)
+FLAGS_DATA(FL, IGNORELASTPLAYPOS,0x00020000)
 // if you move the type mask please edit {Set,Get}ProgramInfoType()
 FLAGS_DATA(FL, TYPEMASK,         0x00F00000)
 FLAGS_DATA(FL, INUSERECORDING,   0x01000000)
diff --git a/mythtv/libs/libmyth/test/test_programinfo/test_programinfo.h b/mythtv/libs/libmyth/test/test_programinfo/test_programinfo.h
index 42945d8a26f..9ba69a7f8de 100644
--- a/mythtv/libs/libmyth/test/test_programinfo/test_programinfo.h
+++ b/mythtv/libs/libmyth/test/test_programinfo/test_programinfo.h
@@ -159,7 +159,7 @@ class TestProgramInfo : public QObject
         {"parttotal", "133"},
         {"playgroup", "Default"},
         {"programflags", "131600"},
-        {"programflags_names", "BOOKMARK|WATCHED|ALLOWLASTPLAYPOS"},
+        {"programflags_names", "BOOKMARK|WATCHED|IGNORELASTPLAYPOS"},
         {"programid", "EP019229360055"},
         {"recenddate", "Wed 26"},
         {"recendtime", "1:02 AM"},
@@ -327,7 +327,7 @@ class TestProgramInfo : public QObject
              RecStatus::Unknown, 19890314, // recordid
              kDupsInAll, kDupCheckSubThenDesc,
              20141007, // findId
-             FL_ALLOWLASTPLAYPOS | FL_WATCHED | FL_BOOKMARK,
+             FL_IGNORELASTPLAYPOS | FL_WATCHED | FL_BOOKMARK,
              AUD_DOLBY,
              VID_DAMAGED | VID_1080 | VID_HDTV,
              SUB_HARDHEAR,
diff --git a/mythtv/libs/libmythtv/mythplayerui.cpp b/mythtv/libs/libmythtv/mythplayerui.cpp
index 08c8cb04425..6f7bdcff513 100644
--- a/mythtv/libs/libmythtv/mythplayerui.cpp
+++ b/mythtv/libs/libmythtv/mythplayerui.cpp
@@ -467,7 +467,7 @@ void MythPlayerUI::EventStart()
             // an actual bookmark and not progstart or lastplaypos information.
             m_playerCtx->m_playingInfo->SetIgnoreBookmark(false);
             m_playerCtx->m_playingInfo->SetIgnoreProgStart(true);
-            m_playerCtx->m_playingInfo->SetAllowLastPlayPos(false);
+            m_playerCtx->m_playingInfo->SetIgnoreLastPlayPos(true);
         }
     }
     m_playerCtx->UnlockPlayingInfo(__FILE__, __LINE__);
diff --git a/mythtv/libs/libmythtv/previewgenerator.cpp b/mythtv/libs/libmythtv/previewgenerator.cpp
index cfd4c3f964d..47005c6a2ba 100644
--- a/mythtv/libs/libmythtv/previewgenerator.cpp
+++ b/mythtv/libs/libmythtv/previewgenerator.cpp
@@ -634,7 +634,7 @@ bool PreviewGenerator::LocalPreviewRun(void)
 {
     m_programInfo.MarkAsInUse(true, kPreviewGeneratorInUseID);
     m_programInfo.SetIgnoreProgStart(true);
-    m_programInfo.SetAllowLastPlayPos(false);
+    m_programInfo.SetIgnoreLastPlayPos(true);
 
     float aspect = 0;
     std::chrono::seconds captime = m_captureTime;
diff --git a/mythtv/libs/libmythtv/tv_play.cpp b/mythtv/libs/libmythtv/tv_play.cpp
index 014ee7c60da..1878ca19856 100644
--- a/mythtv/libs/libmythtv/tv_play.cpp
+++ b/mythtv/libs/libmythtv/tv_play.cpp
@@ -286,7 +286,7 @@ bool TV::StartTV(ProgramInfo* TVRec, uint Flags, const ChannelInfoList& Selectio
         curProgram = new ProgramInfo(*TVRec);
         curProgram->SetIgnoreBookmark((Flags & kStartTVIgnoreBookmark) != 0U);
         curProgram->SetIgnoreProgStart((Flags & kStartTVIgnoreProgStart) != 0U);
-        curProgram->SetAllowLastPlayPos((Flags & kStartTVAllowLastPlayPos) != 0U);
+        curProgram->SetIgnoreLastPlayPos((Flags & kStartTVIgnoreLastPlayPos) != 0U);
     }
 
     // Initialize TV
diff --git a/mythtv/libs/libmythtv/tv_play.h b/mythtv/libs/libmythtv/tv_play.h
index f4da5b295a8..d16adfea463 100644
--- a/mythtv/libs/libmythtv/tv_play.h
+++ b/mythtv/libs/libmythtv/tv_play.h
@@ -109,12 +109,12 @@ enum NoRecorderMsg
 };
 
 enum {
-    kStartTVNoFlags          = 0x00,
-    kStartTVInPlayList       = 0x02,
-    kStartTVByNetworkCommand = 0x04,
-    kStartTVIgnoreBookmark   = 0x08,
-    kStartTVIgnoreProgStart  = 0x10,
-    kStartTVAllowLastPlayPos = 0x20,
+    kStartTVNoFlags           = 0x00,
+    kStartTVInPlayList        = 0x02,
+    kStartTVByNetworkCommand  = 0x04,
+    kStartTVIgnoreBookmark    = 0x08,
+    kStartTVIgnoreProgStart   = 0x10,
+    kStartTVIgnoreLastPlayPos = 0x20,
 };
 
 class AskProgramInfo
diff --git a/mythtv/programs/mythfrontend/playbackbox.cpp b/mythtv/programs/mythfrontend/playbackbox.cpp
index acc6e5af2a8..1b43678d853 100644
--- a/mythtv/programs/mythfrontend/playbackbox.cpp
+++ b/mythtv/programs/mythfrontend/playbackbox.cpp
@@ -2576,11 +2576,11 @@ bool PlaybackBox::Play(
         ignoreProgStart = true;
 
     uint flags =
-        (inPlaylist          ? kStartTVInPlayList       : kStartTVNoFlags) |
-        (underNetworkControl ? kStartTVByNetworkCommand : kStartTVNoFlags) |
-        (!ignoreLastPlayPos  ? kStartTVAllowLastPlayPos : kStartTVNoFlags) |
-        (ignoreProgStart     ? kStartTVIgnoreProgStart  : kStartTVNoFlags) |
-        (ignoreBookmark      ? kStartTVIgnoreBookmark   : kStartTVNoFlags);
+        (inPlaylist          ? kStartTVInPlayList        : kStartTVNoFlags) |
+        (underNetworkControl ? kStartTVByNetworkCommand  : kStartTVNoFlags) |
+        (ignoreLastPlayPos   ? kStartTVIgnoreLastPlayPos : kStartTVNoFlags) |
+        (ignoreProgStart     ? kStartTVIgnoreProgStart   : kStartTVNoFlags) |
+        (ignoreBookmark      ? kStartTVIgnoreBookmark    : kStartTVNoFlags);
 
     playCompleted = TV::StartTV(&tvrec, flags);
 
